src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -37,39 +37,45 @@ class IqPolicyEvaluatorUtil
                                                     final TaskListener listener)
   {
     try {
-      checkArgument(iqPolicyEvaluator.iqStage && iqPolicyEvaluator.iqApplication,
-          'Arguments iqApplication and iqStage are mandatory')
+      String applicationId = ((IqApplication)iqPolicyEvaluator.getIqApplication())?.applicationId
+
+      checkArgument(iqPolicyEvaluator.iqStage && applicationId, 'Arguments iqApplication and iqStage are mandatory')
+
       LoggerBridge loggerBridge = new LoggerBridge(listener)
       loggerBridge.debug(Messages.IqPolicyEvaluation_Evaluating())
 
-      def iqClient = IqClientFactory.getIqClient(new IqClientFactoryConfiguration(
-          credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent, log: loggerBridge))
+      def iqClient = IqClientFactory.getIqClient(
+          new IqClientFactoryConfiguration(credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent,
+              log: loggerBridge))
+
+      def verified = iqClient.verifyOrCreateApplication(applicationId)
+      checkArgument(verified, 'Argument iqApplication ' + applicationId + ' failed verification.')
```

Comment:
Why don't we use the same message we use in the CLI or the bamboo client?

Reply:
> Why don't we use the same message we use in the CLI or the bamboo client?

Changed to 2181041